Sasquatch/Sasquatch/ViewControllers/Sections/CommonSchemaPropertiesTableSection.swift

```diff
@@ -1,94 +1,102 @@
     transmissionTargetSelectorCell?.didSelectTransmissionTarget = reloadSection
   }
   
   override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-    if indexPath.row == 0 {
+    switch indexPath.row {
+    case kTargetSelectorCellRow:
       return transmissionTargetSelectorCell!
-    } else {
-      let cell: MSAnalyticsPropertyTableViewCell? = loadCellFromNib()
-      cell?.valueField.placeholder = "Override value"
-      
-      // Set cell text.
-      var label : String = ""
-      switch indexPath.row {
-      case CommonSchemaProperty.AppName.rawValue:
-        label = "App Name"
-        break
-        
-      case CommonSchemaProperty.AppVersion.rawValue:
-        label = "App Version"
-        break
-        
-      case CommonSchemaProperty.AppLocale.rawValue:
-        label = "App Locale"
-        break
-      default:
-        break
-      }
-      cell!.keyField.text = label
-      
-      let selectedTarget = transmissionTargetSelectorCell?.selectedTransmissionTarget()
-
-      cell!.valueField.text = targets[selectedTarget!]![indexPath.row - 1].1
-      
-      // Set cell to respond to being edited.
-      cell!.valueField.addTarget(self, action: #selector(propertyValueChanged), for: .editingChanged)
-      cell!.valueField.addTarget(self, action: #selector(dismissKeyboard), for: .editingDidEndOnExit)
-      return cell!
+    case kDeviceIdRow:
+        let cell = tableView.dequeueReusableCell(withIdentifier: switchCellIdentifier)!
+        let switcher: UISwitch? = cell.getSubview()
+        let selectedTarget = transmissionTargetSelectorCell?.selectedTransmissionTarget()!
+        switcher!.isOn = collectDeviceIdStates[selectedTarget!]!
+        switcher!.isEnabled = !switcher!.isOn
+        switcher!.addTarget(self, action: #selector(collectDeviceIdSwitchCellEnabled(sender:)), for: .valueChanged)
+        return cell
+    default:
+      let cell = super.tableView(tableView, cellForRowAt: indexPath) as! MSAnalyticsPropertyTableViewCell
+      cell.valueField.placeholder = "Override value"
+      return cell
     }
   }
   
   override func propertyValueChanged(sender: UITextField!) {
     let selectedTarget = transmissionTargetSelectorCell?.selectedTransmissionTarget()
     let propertyIndex = getCellRow(forTextField: sender) - propertyCellOffset()
-    let tableIndex = propertyIndex + 1;
     let target = MSTransmissionTargets.shared.transmissionTargets[selectedTarget!]!
-    targets[selectedTarget!]![propertyIndex].1 = sender.text!
-    
-    if sender.text == nil  || sender.text!.isEmpty {
-      return;
-    }
-    
-    switch tableIndex {
-    case CommonSchemaProperty.AppName.rawValue:
+    propertyValues[selectedTarget!]![propertyIndex] = sender.text!
+    switch propertyIndex {
+    case 0: // App Name.
       target.propertyConfigurator.setAppName(sender.text!)
       break
-    case CommonSchemaProperty.AppVersion.rawValue:
```

Comment:
Why are we now hard-coding these values?  We shouldn't do this if we can avoid it...